src/Stack/Build/ConstructPlan.hs

```diff
@@ -622,15 +622,15 @@ describeConfigDiff config old new
 
 psForceDirty :: PackageSource -> Bool
 psForceDirty (PSLocal lp) = lpForceDirty lp
-psForceDirty (PSUpstream {}) = False
+psForceDirty PSUpstream {} = False
 
 psDirty :: PackageSource -> Maybe (Set FilePath)
 psDirty (PSLocal lp) = lpDirtyFiles lp
-psDirty (PSUpstream {}) = Nothing -- files never change in an upstream package
+psDirty PSUpstream {} = Nothing -- files never change in an upstream package
 
 psLocal :: PackageSource -> Bool
 psLocal (PSLocal _) = True
-psLocal (PSUpstream {}) = False
+psLocal PSUpstream {} = False
```

Comment:
I... didn't know you could do that. I guess this is fine, calling out in case anybody wants to look at this.